### PR TITLE
dotCMS/core#16292 Create DotApiContent class to save content from form

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: projects/dotcms/src/lib/api/DotApiContentType.ts [hunk 1]
```diff
@@ -13,25 +13,64 @@ export class DotApiContentType {
...
+                if (response.status === 200) {
+                    const data = await response.json();
+                    return data.entity;
+                }
+
+                throw <DotCMSError>{
+                    message: await response.text(),
+                    status: response.status
+                };
+            });
```
Comment (freddyucv) [inline projects/dotcms/src/lib/api/DotApiContentType.ts]: I think is more readable something like:

```
                if (response.status !== 200) {
                    throw <DotCMSError>{
                       message: await response.text(),
                       status: response.status
                   };
                }
                const data = await response.json();
                return data.entity;

```

File: projects/dotcms/src/lib/api/DotApiContentType.ts [hunk 2]
```diff
@@ -13,25 +13,64 @@ export class DotApiContentType {
...
+    save<Content extends DotCMSContent>(params: Content): Promise<Response> {
+        return this.dotCMSHttpClient
+            .request({
+                url: `/api/content/save/1`,
+                method: 'POST',
+                body: JSON.stringify(params)
+            })
+            .then(async (response: Response) => {
+                if (response.status === 200) {
+                    return null;
```
Comment (freddyucv) [inline projects/dotcms/src/lib/api/DotApiContentType.ts]: it is weird return null when success, why not return response?

File: projects/dotcms/src/lib/api/DotApiContentType.ts [hunk 3]
```diff
@@ -13,25 +13,64 @@ export class DotApiContentType {
...
+    publish<Content extends DotCMSContent>(params: Content): Promise<Response> {
+        return this.dotCMSHttpClient
+            .request({
+                url: `/api/content/publish/1`,
+                method: 'POST',
+                body: JSON.stringify(params)
+            })
+            .then(async (response: Response) => {
+                if (response.status === 200) {
+                    return null;
```
Comment (freddyucv) [inline projects/dotcms/src/lib/api/DotApiContentType.ts]: same before

File: projects/dotcms/src/lib/api/DotApiContentType.ts (hunk 2 above)
Comment (jdotcms) [reply projects/dotcms/src/lib/api/DotApiContentType.ts]: agree

File: projects/dotcms/src/lib/models/DotCMSContentType.model.ts [hunk 4]
```diff
@@ -38,3 +38,7 @@ export interface DotCMSContentType {
     variable?: string;
     workflow?: string;
 }
+
+export interface DotCMSContent {
+    stName: string;
```
Comment (jdotcms) [inline projects/dotcms/src/lib/models/DotCMSContentType.model.ts]: I would change this to contentTypeName, stName is an ugly legacy name

Comment (alfredo-dotcms) [reply projects/dotcms/src/lib/models/DotCMSContentType.model.ts]: any chance on changing this on the BE? @freddyucv @fmontes

File: projects/dotcms/src/lib/api/DotApiContentType.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply projects/dotcms/src/lib/api/DotApiContentType.ts]: the Endpoint does not return anything

File: projects/dotcms/src/lib/api/DotApiContentType.ts (hunk 3 above)
Comment (alfredo-dotcms) [reply projects/dotcms/src/lib/api/DotApiContentType.ts]: https://github.com/dotCMS/core-web/pull/965#discussion_r270472274

File: projects/dotcms/src/lib/api/DotApiContentType.ts (hunk 2 above)
Comment (fmontes) [reply projects/dotcms/src/lib/api/DotApiContentType.ts]: Should we return the response?

File: projects/dotcms/src/lib/models/DotCMSContentType.model.ts (hunk 4 above)
Comment (fmontes) [reply projects/dotcms/src/lib/models/DotCMSContentType.model.ts]: Not in this issue, we have to modify the endpoint, the documentation etc... way out of scope.

**Pull request merged**